lib/src/select/Listbox.tsx

```diff
@@ -0,0 +1,169 @@
+        >
+          {searchable && (options.length === 0 || !groupsHaveOptions(options)) ? (
+            <OptionsSystemMessage>
+              <NoMatchesFoundIcon>{selectIcons.searchOff}</NoMatchesFoundIcon>
+              No matches found
+            </OptionsSystemMessage>
+          ) : (
+            optional &&
+            !multiple && (
+              <Option
+                id={`option-${0}`}
+                option={optionalItem}
+                onClick={handleOptionOnClick}
+                multiple={multiple}
+                visualFocused={visualFocusIndex === 0}
+                isGroupedOption={false}
+                isLastOption={lastOptionIndex === 0}
+                isSelected={multiple ? currentValue.includes(optionalItem.value) : currentValue === optionalItem.value}
+              />
+            )
+          )}
+          {options.map((option, mapIndex) => {
+            if (option.options) {
+              const groupId = `group-${mapIndex}`;
+              return (
+                option.options.length > 0 && (
+                  <li key={`group-${mapIndex}`}>
+                    <GroupList role="group" aria-labelledby={groupId}>
+                      <GroupLabel role="presentation" id={groupId}>
+                        {option.label}
+                      </GroupLabel>
+                      {option.options.map((singleOption) => {
+                        globalIndex++;
+                        return (
+                          <Option
+                            key={`option-${option.value}`}
+                            id={`option-${globalIndex}`}
+                            option={singleOption}
+                            onClick={handleOptionOnClick}
+                            multiple={multiple}
+                            visualFocused={visualFocusIndex === globalIndex}
+                            isGroupedOption={true}
+                            isLastOption={lastOptionIndex === globalIndex}
+                            isSelected={
+                              multiple ? currentValue.includes(singleOption.value) : currentValue === singleOption.value
+                            }
+                          />
+                        );
+                      })}
+                    </GroupList>
+                  </li>
+                )
+              );
+            } else {
+              globalIndex++;
+              return (
+                <Option
+                  key={`option-${option.value}`}
+                  id={`option-${globalIndex}`}
+                  option={option}
+                  onClick={handleOptionOnClick}
+                  multiple={multiple}
+                  visualFocused={visualFocusIndex === globalIndex}
+                  isLastOption={lastOptionIndex === globalIndex}
+                  isSelected={multiple ? currentValue.includes(option.value) : currentValue === option.value}
+                />
+              );
+            }
+          })}
```

Comment:
I think is better to have this part in a separate function called `mapOptionFunc` as before.

Reply:
Done!